### ISSUE TITLE
Changes to Constant template variables do not persist

### ISSUE THREAD

**Issue by DU-ds** [grafana/grafana#28089]
**What happened**:
changing constant template variables does not persist after save; two fields in the json regresses to previous value on page reload (after saving) causing the old values to be used
**What you expected to happen**:
changing a constant template variable persists in json after reload
**How to reproduce it (as minimally and precisely as possible)**:

1. create new dashboard
2. create constant template variable
3. save
4. change template variable
5. save
6. reload page

**Anything else we need to know?**:

Here's some of the dashboard json.
From between step 3 and 4:
  "templating": {
    "list": [
      {
        "current": {
          "selected": false,
          "text": "val",
          "value": "val"
        },
        "error": null,
        "hide": 2,
        "label": "lab",
        "name": "name",
        "options": [
          {
            "selected": true,
            "text": "val",
            "value": "val"
          }
        ],
        "query": "val",
        "skipUrlSync": false,
        "type": "constant"
      }
    ]
  },

after step 6:

    "templating": {
    "list": [
      {
        "current": {
          "selected": false,
          "text": "val",
          "value": "val"
        },
        "error": null,
        "hide": 2,
        "label": "lab",
        "name": "name",
        "options": [
          {
            "selected": true,
            "text": "newVal",
            "value": "newVal"
          }
        ],
        "query": "newVal",
        "skipUrlSync": false,
        "type": "constant"
      }
    ]
  },

Also, this bug is not consistent. Sometimes it doesn't affect all constant template variables and sometimes it doesn't happen at all. Happens at least 75% of the time for me for my dashboards, causing them to break. I have to go into the json and edit the values. Then they persist when saved. 

**Environment**:
- Grafana version:
noticed this issue since 6.x. Currently on 7.0.3-test.
- Data source type & version:
N/a
- OS Grafana is installed on:
ubuntu
- User OS & Browser:
windows, firefox
- Grafana plugins:
- Others:

**Comment by torkelo** [comment]
Did you check the checkbox "Save current variable values" when you saved the dashboard?

**Comment by torkelo** [comment]
probably a duplicate of #20600

**Comment by grafanabot** [comment]
Thanks for creating this issue! It looks like this has already been reported by another user. We’ve closed this in favor of the existing one. Please consider adding any details you think is missing to that issue.

To avoid having your issue closed in the future, please read our [CONTRIBUTING](https://github.com/grafana/grafana/blob/master/CONTRIBUTING.md) guidelines.

Happy graphing!
